spec/services/repository_service_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RepositoryService, type: :service do
+  let(:user) { create(:user) }
+  let(:service) { described_class.new(user) }
+
+  before do
+    mock_repos = [
+      { 'name' => 'Repo1', 'owner' => { 'login' => 'user1' }, 'updated_at' => '2023-11-07' },
```

Comment:
to powinna byc headless factory / fixture

Reply:
albo uzyj VCR

Reply:
użyłem webmocka ;)